### PR TITLE
RegEx Assistant "tuneups"

### PR DESCRIPTION
**Pull request by Vogel612**
partially incorporated changes suggested in reviews, fixed prior design mistake

### CODE REVIEW

**Comment by Vogel612** [comment]
Please check and merge :)

File: Rubberduck.RegexAssistant/IRegularExpression.cs [hunk 1]
```diff
@@ -8,19 +9,21 @@ namespace Rubberduck.RegexAssistant
...
     public class ConcatenatedExpression : IRegularExpression
     {
-        private readonly Quantifier _quantifier;
         private readonly IList<IRegularExpression> _subexpressions;
 
         public ConcatenatedExpression(IList<IRegularExpression> subexpressions)
         {
+            if (subexpressions == null)
+            {
+                throw new NullReferenceException();
```
Comment (retailcoder) [inline Rubberduck.RegexAssistant/IRegularExpression.cs]: Shouldn't that be an `ArgumentNullException`?

File: Rubberduck.RegexAssistant/IRegularExpression.cs [hunk 2]
```diff
@@ -50,13 +45,16 @@ public IList<IRegularExpression> Subexpressions
...
     public class AlternativesExpression : IRegularExpression
     {
-        private readonly Quantifier _quantifier;
         private readonly IList<IRegularExpression> _subexpressions;
 
         public AlternativesExpression(IList<IRegularExpression> subexpressions)
         {
+            if (subexpressions == null)
+            {
+                throw new NullReferenceException();
```
Comment (retailcoder) [inline Rubberduck.RegexAssistant/IRegularExpression.cs]: ditto

File: Rubberduck.RegexAssistant/IRegularExpression.cs [hunk 3]
```diff
@@ -87,27 +77,22 @@ public IList<IRegularExpression> Subexpressions
...
-            _quantifier = quantifier;
-        }
-
-        public string Description
-        {
-            get
+            if (atom == null)
             {
-                return string.Format("{0} {1}.", Atom.Description, Quantifier.HumanReadable());
+                throw new NullReferenceException();
```
Comment (retailcoder) [inline Rubberduck.RegexAssistant/IRegularExpression.cs]: aaand here too

**Comment by retailcoder** [comment]
So, that's all just implementing the CR recommendations, right?

The only thing I don't like is throwing a `NullReferenceException` when the null object is a parameter - that should be an `ArgumentNullException`.

**Pull request merged**